File: app/src/main/java/ua/dp/rename/dniprostreets/api/RenamedObjectsService.java
package ua.dp.rename.dniprostreets.api;

import retrofit.http.GET;
import rx.Observable;
import ua.dp.rename.dniprostreets.entity.ApiDataHolder;
import ua.dp.rename.dniprostreets.entity.LastUpdateHolder;

public interface RenamedObjectsService {

    @GET("/rename.json")
    Observable<ApiDataHolder> getJson();

    @GET("/last_data_update.json")
    Observable<LastUpdateHolder> getLastDataUpdateTimestamp();
}

File: app/src/main/java/ua/dp/rename/dniprostreets/api/RenamedObjectsService.kt
package ua.dp.rename.dniprostreets.api

import io.reactivex.Single
import retrofit2.Response
import retrofit2.http.GET
import ua.dp.rename.dniprostreets.entity.ApiDataHolder
import ua.dp.rename.dniprostreets.entity.LastUpdateHolder

interface RenamedObjectsService {

   @GET("rename.json")
   fun getJson(): Single<Response<ApiDataHolder>>

   @GET("last_data_update.json")
   fun getLastDataUpdateTimestamp(): Single<Response<LastUpdateHolder>>
}
